msticpy/data/azure_blob_storage.py

```diff
@@ -0,0 +1,260 @@
+from azure.core.exceptions import (
+    ResourceExistsError,
+    ResourceNotFoundError,
+    ServiceRequestError,
+)
+from azure.storage.blob import BlobServiceClient, generate_blob_sas
+
+from ..common.azure_auth import az_connect
+from ..common.azure_auth_core import AzCredentials
+
+
+class AzureBlobStorage:
+    """Class for interacting with Azure Blob Storage."""
+
+    def __init__(self, abs_name: str, connect: bool = False):
+        """Initialize connector for Azure Python SDK."""
+        self.connected = False
+        self.abs_site = f"{abs_name}.blob.core.windows.net"
+        self.credentials: Optional[AzCredentials] = None
+        self.abs_client: Optional[BlobServiceClient] = None
+        if connect is True:
+            self.connect()
+
+    def connect(
+        self,
+        auth_methods: List = None,
+        silent: bool = False,
+    ):
+        """Authenticate with the SDK."""
+        self.credentials = az_connect(auth_methods=auth_methods, silent=silent)
+        if not self.credentials:
+            raise CloudError("Could not obtain credentials.")
+        self.abs_client = BlobServiceClient(self.abs_site, self.credentials.modern)
+        if not self.abs_client:
+            raise CloudError("Could not create a Blob Storage client.")
+        self.connected = True
+
+    def containers(self) -> pd.DataFrame:
+        """Return containers in the Azure Blob Storage Account."""
+        try:
+            container_list = self.abs_client.list_containers()  # type:ignore
+        except ServiceRequestError as err:
+            raise CloudError(
+                "Unable to connect check the Azure Blob Store account name"
+            ) from err
+        if container_list:
+            containers_df = _parse_returned_items(
+                container_list, remove_list=["lease", "encryption_scope"]
+            )
+        else:
+            containers_df = None
+        return containers_df
+
+    def create_container(self, container_name: str, **kwargs) -> pd.DataFrame:
+        """
+        Create a new container within the Azure Blob Storage account.
+
+        Parameters
+        ----------
+        container_name : str
+            The name for the new container.
+        Additional container parameters can be passed as kwargs
+
+        Returns
+        -------
+        pd.DataFrame
+            Details of the created container.
+
+        """
+        try:
+            new_container = self.abs_client.create_container(  # type: ignore
+                container_name, **kwargs
+            )  # type:ignore
+        except ResourceExistsError as err:
+            raise CloudError(f"Container {container_name} already exists.") from err
+        properties = new_container.get_container_properties()
+        container_df = _parse_returned_items(
+            [properties], ["encryption_scope", "lease"]
+        )
+        return container_df
+
+    def blobs(self, container_name: str) -> pd.DataFrame:
```

Comment:
Since you could return None, this will prob generate a mypy error - return type should be Optional[pd.DataFrame